src/ru/nsu/ccfit/zuev/osu/menu/ModMenu.java

```diff
@@ -129,26 +192,48 @@ public void init() {
 
         //line 1
         addButton(offset, Config.getRES_HEIGHT() / 2 - button.getHeight() * 3, "selection-mod-easy", GameMod.MOD_EASY);
-        addButton(offset + offsetGrowth, Config.getRES_HEIGHT() / 2 - button.getHeight() * 3, "selection-mod-nofail", GameMod.MOD_NOFAIL);
-        addButton(offset + offsetGrowth * 2, Config.getRES_HEIGHT() / 2 - button.getHeight() * 3, "selection-mod-halftime", GameMod.MOD_HALFTIME);
-        addButton(offset + offsetGrowth * 3, Config.getRES_HEIGHT() / 2 - button.getHeight() * 3, "selection-mod-reallyeasy", GameMod.MOD_REALLYEASY);
+
+        // Used to define the X offset of each button according to its visibility
+        int factor = 1;
+
+        if (!Multiplayer.isMultiplayer)
+            addButton(offset + offsetGrowth * factor++, Config.getRES_HEIGHT() / 2 - button.getHeight() * 3, "selection-mod-nofail", GameMod.MOD_NOFAIL);
```

Comment:
NF should be usable in multiplayer. In the current state of multiplayer, it should allow online score submission regardless of fail status.

In the future, when we support displaying failed scores (that did not recover in time), NF would also prevent the player from going to failed state.

Reply:
But what is the purpose of NF in multiplayer?

Reply:
> In the current state of multiplayer, it should allow online score submission regardless of fail status.

> In the future, when we support displaying failed scores (that did not recover in time), NF would also prevent the player from going to failed state.